Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/Parameters/SqlTableValuedDataParameterDefinitionFactoryTests.cs

```diff
@@ -0,0 +1,98 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+#nullable enable
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model.Building;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Parameters;
+using Remotion.Data.DomainObjects.Queries;
+using Remotion.Data.DomainObjects.UnitTests.Factories;
+using Remotion.Development.UnitTesting;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.SqlServer.Parameters;
+
+[TestFixture]
+public class SqlTableValuedDataParameterDefinitionFactoryTests
```

Comment:
I'm missing the tests for ISet<T> and IReadOnlySet<T> and IReadOnlyCollection<T> and ICollection<T>.

The existing test for Array should remain.

I wonder if we should have integration tests for some of the derived collection types.

Reply:
I have refactored the tests into a single method with `[TestCaseSource]`, and added cases for `ReadOnlyCollection<int>`, `ArrayList`, and `List<int>`, as well as Mocks for the collection and set interfaces.